include/mbedtls/x509_crt.h

```diff
@@ -96,6 +96,8 @@ typedef struct mbedtls_x509_crt
     mbedtls_pk_type_t sig_pk;           /**< Internal representation of the Public Key algorithm of the signature algorithm, e.g. MBEDTLS_PK_RSA */
     void *sig_opts;             /**< Signature options to be passed to mbedtls_pk_verify_ext(), e.g. for RSASSA-PSS */
 
+    uint32_t allowed_unsupported_critical_exts; /**< Optional Bit flags which represent runtime-enabled unsupported critical extensions, e.g. MBEDTLS_X509_EXT_NAME_CONSTRAINTS */
```

Comment:
This looks like it should be an argument to the parser, passed via the profile (`mbedtls_x509_crt_profile`). Why would the certificate object contain which extensions are _allowed_, rather than which extensions are _present_?

Reply:
Sure.